### DISCUSSION TITLE
Help changing an output filename

### DISCUSSION THREAD

**Discussion by PauloEgidio** [ImageMagick/ImageMagick#7314] [Help] [answered]
### ImageMagick version

7

### Operating system, version and so on

Windows

### Description

I am using negate to invert colors: "magick convert image1.png -channel RGB -negate image1_white.png". It works perfectly.
But I want to do it batch and 
magick convert *.png -set filename:fname %t -channel RGB -negate "%[filename:fname]_white.png" 
gives an error 3359.
can someone help me please or kindly point me to an output filename documentation page? with *.something as input and changing filenames?
Thanks!

**Comment by fmw42** [comment, marked as answer]
In IM 7, use magick, not magick convert. **Doing that works for me**.  You were using IM 6 syntax by adding convert.  You need IM 7 syntax.  So remove "convert"

Or you could use magick mogrify

```
change directory to your_image_directory
magick mogrify -format png -path your_output_directory -negate -channel rgb -negate *.png
```

See https://imagemagick.org/Usage/basics/#mogrify

**Reply by PauloEgidio** [reply]
Work perfectly! Many thanks!

**Reply by fmw42** [reply]
See changes in the above.
